qw5q_gold_qblox.py

```diff
@@ -157,85 +34,82 @@ def create(runcard_path=RUNCARD):
         runcard_path (str): Path to the runcard file.
     """
 
-    def instantiate_module(modules, cls, name, address, settings):
-        module_settings = settings[name]
-        modules[name] = cls(name=name, address=address, settings=module_settings)
-        return modules[name]
-
+    runcard = load_runcard(runcard_path)
     modules = {}
 
     cluster = Cluster(
         name="cluster",
-        address="192.168.0.20",
-        settings=instruments_settings["cluster"],
+        address=ADDRESS,
+        settings=Cluster_Settings(reference_clock_source=ReferenceClockSource.INTERNAL),
     )
 
-    qrm_rf_a = instantiate_module(
-        modules, ClusterQRM_RF, "qrm_rf_a", "192.168.0.20:16", instruments_settings
-    )  # qubits q0, q1, q5
-    qrm_rf_b = instantiate_module(
-        modules, ClusterQRM_RF, "qrm_rf_b", "192.168.0.20:18", instruments_settings
-    )  # qubits q2, q3, q4
-
-    qcm_rf0 = instantiate_module(
-        modules, ClusterQCM_RF, "qcm_rf0", "192.168.0.20:6", instruments_settings
-    )  # qubit q0
-    qcm_rf1 = instantiate_module(
-        modules, ClusterQCM_RF, "qcm_rf1", "192.168.0.20:8", instruments_settings
-    )  # qubits q1, q2
-    qcm_rf2 = instantiate_module(
-        modules, ClusterQCM_RF, "qcm_rf2", "192.168.0.20:10", instruments_settings
-    )  # qubits q3, q4
-
-    qcm_bb0 = instantiate_module(
-        modules, ClusterQCM_BB, "qcm_bb0", "192.168.0.20:2", instruments_settings
-    )  # qubit q0
-    qcm_bb1 = instantiate_module(
-        modules, ClusterQCM_BB, "qcm_bb1", "192.168.0.20:4", instruments_settings
-    )  # qubits q1, q2, q3, q4
-
     # DEBUG: debug folder = report folder
     # import os
     # folder = os.path.dirname(runcard) + "/debug/"
     # if not os.path.exists(folder):
     #     os.makedirs(folder)
     # for name in modules:
     #     modules[name]._debug_folder = folder
+    qcm_bb0 = ClusterQCM_BB("qcm_bb0", f"{ADDRESS}:2", cluster)
+    qcm_bb1 = ClusterQCM_BB("qcm_bb1", f"{ADDRESS}:4", cluster)
+    qcm_rf0 = ClusterQCM_RF("qcm_rf0", f"{ADDRESS}:6", cluster)
+    qcm_rf1 = ClusterQCM_RF("qcm_rf1", f"{ADDRESS}:8", cluster)
+    qcm_rf2 = ClusterQCM_RF("qcm_rf2", f"{ADDRESS}:10", cluster)
+    qrm_rf_a = ClusterQRM_RF("qrm_rf_a", f"{ADDRESS}:16", cluster)
+    qrm_rf_b = ClusterQRM_RF("qrm_rf_b", f"{ADDRESS}:18", cluster)
```

Comment:
Just to tell: in principle this could be also generated, rather than enumerated:
```suggestion
	clusters = {name: ClusterQCM_BB(name, f"{ADDRESS}:" + str(2*(i+1)), cluster) for i, name in enumerate([...])}
```

I believe, only in py>=3.12, you could write:
```suggestion
	clusters = {name: ClusterQCM_BB(name, f"{ADDRESS}:{2*(i+1)}", cluster) for i, name in enumerate([...])}
```

Reply:
Yes but this is true only if, as in this case, you connect the modules in the even slots of the cluster. Often they are connected pretty randomly, so it must be taken into account the possibility of random or not so even connection

Reply:
I'm aware that some of these simplifications are working only in specific setups. That's why the people in (or close to) the lab should provide feedback about them.

Moreover, I made an error, since there are two different cluster classes involved (`_RF` and `_BB`).

However, the information here could be slightly compressed anyhow. But most likely it's not convenient in this case.
In case it will be in the future, we could reconsider :)